### PR TITLE
Add note about middleware order

### PR DESCRIPTION
**Pull request by JD-Robbs**

### CODE REVIEW

**Comment by JD-Robbs** [comment]
Right, following this documentation change means breaking the default base route "/" redirecting to the default/detected locale. 

However, without following this, the `lang` param is never set in the router, meaning that `I18n::locale()` is always called with the default locale and not the one that is actually detected from the URL. 

Hence, I'm not quite sure if there's a bug, or if one is supposed to call the middleware twice?

File: README.md [hunk 1]
```diff
@@ -115,7 +115,9 @@ $middlware->add(new \ADmad\I18n\Middleware\I18nMiddleware([
 ]));
 ```
 
-The keys of `languages` array are the language prefixes you use in your URL.
+The keys of `languages` array are the language prefixes you use in your URL. 
+
+To ensure that the `lang` router param is available, you must add this middleware *after* adding CakePHP's default routing middleware (i.e. after `->add(new RoutingMiddleware($this))`).
```
Comment (ADmad) [inline README.md]: `RoutingMiddleware` doesn't need an argument `$this`.

**Comment by ADmad** [comment]
You are only supposed to include to add the middleware once after `RoutingMiddleware` and redirecting to lang URL when accessing `/` works just fine.

**Comment by JD-Robbs** [comment]
Thanks, I guess something strange appears to be going on in my code.  I get a `MissingRouteException` when accessing "/" - unless I also add `I18nMiddleware` before `RoutingMiddleware`.

**Comment by ADmad** [comment]
You should setup a route for `/` in your app routes.

**Comment by JD-Robbs** [comment]
That's the odd thing - I have.

Anyhow, I've copied the respective redirection code into a separate middleware and am now adding that before the routing middleware - rather than adding the same middleware twice.

```
<?php

namespace App\Middleware;

use ADmad\I18n\Middleware\I18nMiddleware;
use Cake\Core\InstanceConfigTrait;
use Psr\Http\Message\ResponseInterface;
use Psr\Http\Message\ServerRequestInterface;
use Zend\Diactoros\Response\RedirectResponse;

class I18nRedirectMiddleware
{
    use InstanceConfigTrait;

    protected $_defaultConfig = [
        'defaultLanguage' => 'en_US'
    ];

    public function __construct($config = [])
    {
        $this->setConfig($config);
    }

    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, $next)
    {
        $config = $this->getConfig();
        $url    = $request->getUri()->getPath();

        if ($url === '/') {
            $lang       = $config['defaultLanguage'];
            $statusCode = 302;
            $lang       = $this->detectLanguage($request, $lang);

            $response = new RedirectResponse(
                $request->getAttribute('webroot') . $lang,
                $statusCode
            );

            return $response;
        }

        return $next($request, $response);
    }

    public function detectLanguage(ServerRequestInterface $request, $default = null)
    {
        return (new I18nMiddleware())->detectLanguage($request, $default);
    }
}
```

**Pull request merged**